src/routes/auth.go

```diff
@@ -0,0 +1,39 @@
+package routes
+
+import (
+	"github.com/dana-team/platform-backend/src/types"
+	"github.com/dana-team/platform-backend/src/utils"
+	"github.com/gin-gonic/gin"
+	"net/http"
+)
+
+// Login handles user authentication and issues a token on successful login.
+// It expects a JSON payload containing the user's credentials:
+//
+//	{
+//	   "username": "your_username", "password": "your_password"
+//	}
+//
+// It responds with a JSON object containing the token on successful login:
+//
+//	{
+//	  "token": "your_generated_token"
+//	}
+//
+// If there's an error during authentication, it responds with an appropriate error message.
+func Login() gin.HandlerFunc {
+	return func(c *gin.Context) {
+		var user types.User
+		if err := c.ShouldBindJSON(&user); err != nil {
+			c.JSON(http.StatusBadRequest, gin.H{"error": "Invalid request payload"})
+			return
+		}
+
+		token, err := utils.ObtainOpenshiftToken(user.Username, user.Password)
+		if err != nil {
+			c.JSON(http.StatusUnauthorized, gin.H{"error": "Invalid credentials"})
```

Comment:
Please add `return` to stop execution of this and also use `c.AbortWithStatusJSON(...)`.

Reply:
I have already added a return at line 44. Is there anything else you'd suggest?

Reply:
Nope, thanks! :)